src/test/java/org/openmrs/contrib/qaframework/automation/ClinicalVisitSteps.java

```diff
@@ -119,7 +120,7 @@ public void loadAllergiesPage() {
 
 	@Then("the system loads Allergies board page")
 	public void systemLoadsAllergiesPage() {
-		assertTrue(textExists("Allergies"));
+		assertPage(allergyPage.waitForPage());
```

Comment:
True , i think this approach have another limitation, i don't think we don't need to include waiting element every time.

Reply:
assertPage() usually applies its implementation faster and the pages take longer to load especially on firefox engines. I would think we may not get rid of `waitForPage()` function in such assertions.

Reply:
@sherrif10  could you please refer this conversation we has some time [here ](https://github.com/openmrs/openmrs-contrib-qaframework/pull/153#discussion_r749389133) about this issue.

Reply:
> 

@kdaud  this is what am trying to stretch.Thanks for this point.

Reply:
> 
This comment seems you are supporting we leave in assertTexts, let me know if it's clear

Reply:
@sherrif10 which comment? And supporting what?